bees/sunbee/sunbee.go

```diff
@@ -36,19 +36,37 @@ import (
 // with writing your own Bees.
 type SunBee struct {
 	bees.Bee
-	city string
+	query    string
+	offset   int64
+	lat, lon float64
 }
 
 // Run executes the Bee's event loop.
 func (mod *SunBee) Run(eventChan chan bees.Event) {
 	gominatim.SetServer("https://nominatim.openstreetmap.org/")
+	qry := gominatim.SearchQuery{
+		Q: mod.query,
+	}
+	resp, err := qry.Get()
+	if err != nil {
+		mod.LogFatal("Error geocoding %s. err: %v", mod.query, err)
+	}
+
+	mod.lat, err = strconv.ParseFloat(resp[0].Lat, 64)
+	if err != nil {
+		mod.LogFatal("failed parsing latitude from response. err: %v", err)
+	}
+	mod.lon, err = strconv.ParseFloat(resp[0].Lon, 64)
+	if err != nil {
+		mod.LogFatal("failed parsing longitude from response. err: %v", err)
+	}
 
 	for {
 		select {
 		case <-mod.SigChan:
 			return
-		case <-time.After(time.Duration(1 * time.Minute)):
-			mod.check(mod.city, eventChan)
+		case <-time.After(time.Duration(10 * time.Second)):
```

Comment:
Isn't this a bit excessive?

Reply:
Ignore that comment, just saw you already fixed that in a later commit and my GitHub diff just wasn't up-to-date.